src/twisted/internet/base.py

```diff
@@ -1168,6 +1168,10 @@ def getDestination(self):
             reflect.qual(self.__class__) + " did not implement "
             "getDestination")
 
+    def __repr__(self):
+        return "<%s instance at %x %s>" % (
```

Comment:
I think that the `%x` part should be `0x%x` as object IDs are generally presented with that prefix but `%x` doesn't automatically add it:

```python
>>> object()
<object object at 0x7f5894500080>
>>> '%x' % id(object())
'7f58945000a0'
```